src/open_inwoner/accounts/signals.py

```diff
@@ -37,27 +36,26 @@ def update_user_from_klant_on_login(sender, user, request, *args, **kwargs):
         return
 
     # OpenKlant2
-    # TODO: replace with proper config and refactor branching
-    use_ok2 = getattr(settings, "OPENKLANT2_ACTIVE", None)
-    if use_ok2 and (openklant2_config := OpenKlant2Config.from_django_settings()):
-        try:
-            service = OpenKlant2Service(config=openklant2_config)
-        except RuntimeError:
-            logger.error("OpenKlant2 service failed to build")
-            return
-
-        if not (fetch_params := service.get_fetch_parameters(request=request)):
-            return
-
-        partij, created = service.get_or_create_partij_for_user(
-            fetch_params=fetch_params, user=user
-        )
-        if partij and not created:
-            service.update_user_from_partij(partij_uuid=partij["uuid"], user=user)
+    try:
+        service = OpenKlant2Service()
+    except ImproperlyConfigured:
+        logger.error("OpenKlant2 configuration missing")
+    except RuntimeError:
+        logger.error("Failed to build OpenKlant2Service")
+    else:
+        if fetch_params := service.get_fetch_parameters(request=request):
+            partij, created = service.get_or_create_partij_for_user(
+                fetch_params=fetch_params, user=user
+            )
+            if partij and not created:
+                service.update_user_from_partij(partij_uuid=partij.uuid, user=user)
```

Comment:
Can't skip the `else` block for the first service because we wouldn't be able to do `service.get_fetch_parameters`. I did export the code in the `else` block to a separate function.